docs/topics/downloader-middleware.rst

```diff
@@ -922,13 +922,118 @@ RobotsTxtMiddleware
+Python's inbuilt ``robots.txt`` parser. The parser is fully compliant with `Martijn Koster's 
+1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_, with support for
+``Crawl-delay`` and ``Request-rate`` directives also available from Python 3.6. It lacks
+support for wildcard matching. Scrapy uses this parser by default.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.PythonRobotParser``
+
+.. _rerp-parser:
+
+Robotexclusionrulesparser
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Robotexclusionrulesparser <http://nikitathespider.com/python/rerp/>`_ is fully compliant
+with `Martijn Koster's 1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_,
+with support for wildcard matching, ``Crawl-delay`` directive, and ``Sitemap`` directive.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.RerpRobotParser``
+
+.. _reppy-parser:
+
+Reppy parser
+~~~~~~~~~~~~
+
+`Reppy <https://github.com/seomoz/reppy/>`_ is a Python wrapper around `Robots Exclusion
+Protocol Parser for C++ <https://github.com/seomoz/rep-cpp>`_. The parser is fully compliant
+with `Martijn Koster's 1996 draft specification <http://www.robotstxt.org/norobots-rfc.txt>`_,
+with support for wildcard matching, ``Crawl-delay`` directive, and ``Sitemap`` directive.
+Unlike `RobotFileParser <https://docs.python.org/3.7/library/urllib.robotparser.html>`_ and 
+`Robotexclusionrulesparser <http://nikitathespider.com/python/rerp/>`_, it uses the length based 
+rule, in particular for ``Allow`` and ``Disallow`` directives, where the most specific rule based 
+on the length of the path trumps the less specific (shorter) rule.
+
+In order to use this parser, set:
+
+* :setting:`ROBOTSTXT_PARSER` to ``scrapy.extensions.robotstxtparser.ReppyRobotParser``
+
+.. _support-for-new-robots-parser:
+
+Implementing support for a new parser
+-------------------------------------
+
+You can implement support for a new robots.txt_ parser by implementing the
+interface described below.
+
+.. class:: scrapy.extensions.robotstxtparser.RobotsParser
+
+    .. method:: from_crawler(crawler, content)
+
+      Parse the content of a robots.txt_ file as bytes. This must be a class method.
+      It must return a new instance of the parser backend.
+
+      :param crawler: crawler which made the request
+      :type crawler: :class:`~scrapy.crawler.Crawler` instance
+      
+      :param content: Content of a robots.txt_ file.
+      :type content: bytes
+
+    .. method:: allowed(url, user_agent)
+
+      Return ``True`` if  ``user_agent`` is allowed to crawl ``url``, otherwise return ``False``.
+
+      :param url: Absolute URL
+      :type url: string
+
+      :param user_agent: User agent
+      :type user_agent: string
+
+    .. method:: sitemaps()
+
+      Return a generator yielding URL to sitemaps on the website. If there is no sitemap specified, return an empty generator.
```

Comment:
we're not using this method anywhere, right?

Reply:
Currently No, but we do have plans to to make `scrapy.spiders.SitemapSpider` use this interface.

Reply:
@anubhavp28 Maybe we can remove `sitemaps()` from this patch and, once this is merged, work on a second patch that implements this and the integration with `SitemapSpider`.

Reply:
Yes, I do think that will be better for both `sitemaps` and `crawl-delay` since we are not using it anywhere currently. Also, keeping these two functions in interface reference could give the impression that only parsers with support for `Crawl-delay` and `Sitemaps` can work with Scrapy.

Reply:
> @anubhavp28 Maybe we can remove `sitemaps()` from this patch and, once this is merged, work on a second patch that implements this and the integration with `SitemapSpider`.

Are you suggesting to remove `sitemaps` mentions from the documentation or to remove `sitemaps` function from code? I think we could just remove `sitemaps` mentions in documentation while keeping the code same.

Reply:
I was actually suggesting both, to keep the changes as clean as possible, i.e. do not add any useless code.

Keep the changes somewhere you can recover them easily, though, e.g. a local branch or text file somewhere. As soon as these changes are merged, it would be great for you to start working on a patch for sitemap support.